src/Parser/DemonHunter/Havoc/Modules/Traits/CriticalChaos.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Analyzer from 'Parser/Core/Analyzer';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+import CritEffectBonus from 'Parser/Core/Modules/Helpers/CritEffectBonus';
+
+const CRITICAL_CHAOS_INCREASE = 0.06;
+
+const CRITICAL_CHAOS_SPELLS = [
+	SPELLS.CHAOS_STRIKE_MH_DAMAGE.id,
+	SPELLS.CHAOS_STRIKE_OH_DAMAGE.id,
+	SPELLS.ANNIHILATION_MH_DAMAGE.id,
+	SPELLS.ANNIHILATION_OH_DAMAGE.id,
+];
+/*
+* Critical Chaos (Artifact Trait)
+* Increases critical strike damage dealt by Chaos Strike by 6%.
+*/
+
+class CrticalChaos extends Analyzer {
+	static dependencies = {
+		combatants: Combatants,
+		critEffectBonus: CritEffectBonus,
+	}
+
+	rank = 0;
+	damage = 0;
+
+	on_initialized() {
+ 		this.rank = this.combatants.selected.traitsBySpellId[SPELLS.CRITICAL_CHAOS.id];
+ 		this.active = this.rank > 0;
+
+ 		if (this.active) {
+      this.critEffectBonus.hook(this.getCritEffectBonus.bind(this));
+    }
+ 	}
+
+ 	getCritEffectBonus(critEffectModifier, event) {
+ 		const spellId = event.ability.guid;
+    if (CRITICAL_CHAOS_SPELLS.includes(spellId)) {
+      critEffectModifier += this.rank * CRITICAL_CHAOS_INCREASE;
+    }
+    return critEffectModifier;
+  }
+
+  on_byPlayer_damage(event) {
+  	const spellId = event.ability.guid;
+  	if (!CRITICAL_CHAOS_SPELLS.includes(spellId) || event.hitType !== HIT_TYPES.CRIT) {
+  		return;
+  	}
+
+  	const raw = (event.amount || 0) + (event.absorbed || 0);
+  	const rawNormalPart = raw / this.critEffectBonus.getBonus(event);
```

Comment:
Right now you're getting the non crit damage (which matches the var name), I assume you'd want only the crit bonus damage instead? `critBonusDamage = raw - raw / this.critEffectBonus.getBonus(event);`